packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +499,857 @@ describe('Testing authentication module', () => {
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe(errorMessage);
+          });
+        } else {
+          it(`authentication.notifyFailure should not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => authentication.notifyFailure()).toThrow(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
     });
   });
 
-  describe('Testing authentication.notifyFailure', () => {
-    const allowedContexts = [FrameContexts.authentication];
-    it('authentication.notifyFailure should not allow calls before initialization', () => {
-      expect(() => authentication.notifyFailure()).toThrowError('The library has not yet been initialized');
+  describe('FRAMELESS - authentication tests', () => {
+    beforeEach(() => {
+      // Use to send a mock message from the app.
+      framelessPostMock.messages = [];
+      // Set a mock window for testing
+      app._initialize(framelessPostMock.mockWindow);
     });
 
-    it('should not close auth window before notify failure message has been sent', async () => {
-      expect.assertions(5);
-      const closeWindowSpy = jest.spyOn(utils.mockWindow, 'close');
-
-      const initPromise = app.initialize();
-      const initMessage = utils.findMessageByFunc('initialize');
-      expect(initMessage).not.toBeNull();
-
-      authentication.notifyFailure(errorMessage);
-      let message = utils.findMessageByFunc('authentication.authenticate.failure');
-      expect(message).toBeNull();
-      expect(closeWindowSpy).not.toHaveBeenCalled();
-
-      utils.respondToMessage(initMessage, 'authentication');
-      await initPromise;
-      message = utils.findMessageByFunc('authentication.authenticate.failure');
-      expect(message).not.toBeNull();
-
-      // Wait 100ms for the message queue and 200ms for the close delay
-      await new Promise<void>(resolve =>
-        setTimeout(() => {
-          expect(closeWindowSpy).toHaveBeenCalled();
-          resolve();
-        }, 350),
-      );
+    afterEach(() => {
+      // Reset the object since it's a singleton
+      if (app._uninitialize) {
+        app._uninitialize();
+      }
     });
 
-    Object.values(FrameContexts).forEach(context => {
-      if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
-        it(`authentication.notifyFailure should successfully notify auth failure ${context} context`, async () => {
-          await utils.initializeWithContext('authentication');
+    describe('Testing authentication.initialize function', () => {
+      it('authentication.initialize should successfully register authentication.authenticate.success/failure handler', async () => {
```

Comment:
I think this is literally identical to the framed version of the test since you don't use the mock at all

Reply:
Nice catch!